src/python/py/models/builder.py

```diff
@@ -2521,7 +2595,7 @@ def make_model(self, input_path):
         else:
             # Load PyTorch model
             extra_kwargs = {"num_hidden_layers": self.num_layers} if "num_hidden_layers" in self.extra_options else {}
-            model = AutoModelForCausalLM.from_pretrained(self.model_name_or_path, cache_dir=self.cache_dir, token=self.hf_token, trust_remote_code=True, **extra_kwargs)
+            model = AutoModelForCausalLM.from_pretrained(self.model_name_or_path, cache_dir=self.cache_dir, token=self.hf_token, trust_remote_code=True, torch_dtype="auto", **extra_kwargs)
```

Comment:
@kunal-vaishnavi I wonder if it's possible to set torch_dtype properly to reduce some need for casting?